### ISSUE TITLE
Combined/additive transport from multiple transport models in different radial ranges

### ISSUE THREAD

**Issue by jcitrin** [google-deepmind/torax#550]

**Comment by demoncoder-crypto** [comment]
Hey, i would love to work on this issue, can you pls assign me

**Comment by jcitrin** [comment]
@theo-brown I believe you made this feature request a while back. Could you provide a more detailed specification of how the feature should look, which will then help with the design?

**Comment by jcitrin** [comment]
@demoncoder-crypto yes very possibly, but let's first understand the scope and discuss the design. This may also be a feature that should wait post-new-release when the config patterns stabilize.

**Comment by theo-brown** [comment]
Thanks for your interest @demoncoder-crypto! I think it might be easiest if I take this on as I've got a clearer idea of the use case.

The basic idea is as follows.

#### Motivation
Some transport models accurately model turbulent transport in one part of the plasma but not in another. This is particularly the case for STEP when using models designed for conventional aspect ratio tokamaks - QLKNN, for example, does an OK job at modelling STEP turbulent transport off axis but gets progressively worse as you move towards the axis. In our JETTO runs, we've got round this by combining QLKNN with small amounts of Rechester-Rosenbluth or BgB.

#### Proposed solution
- New `CombinedTransportModel` class subclassing `TransportModel`, initialised with a list of other transport models and radial ranges.
- Call implementation of this class calls the child models and adds them based on their region of validity. This means that smoothing and clipping will be performed on the sum model, rather than the individual ones, which I think is desirable - it'll help smooth over discontinuities from restricting the models to a range. There's also potential for doing something more clever here.
- Implement a new pydantic model for this class, making sure the build function correctly constructs all the children.

I haven't yet got up to speed with the new config system, but I envisage this being set as something like
```python
'transport_model': {
    'combined': {
        'bohm-gyrobohm': [0, 0.5],
       'qlknn': [0.3, 1.0],
    }
}
```

**Comment by theo-brown** [comment]
@jcitrin if you're happy with this plan then I can make a PR

**Comment by jcitrin** [comment]
Yes please go ahead with this one, thanks! Should discuss the design further.
